src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -253,30 +253,50 @@ public class Memory {
     }
 
     /// <summary>
-    ///     Allow a class to register for a certain memory range.
+    /// Allow a class to register for a certain memory range.
     /// </summary>
     /// <param name="baseAddress">The start of the frame</param>
     /// <param name="size">The size of the window</param>
     /// <param name="memoryDevice">The memory device to use</param>
     public void RegisterMapping(uint baseAddress, uint size, IMemoryDevice memoryDevice) {
         uint endAddress = baseAddress + size;
-        if (endAddress >= _memoryDevices.Length) {
-            Array.Resize(ref _memoryDevices, (int)endAddress);
+        if (endAddress >= _memoryDevices.Count) {
+            long length = endAddress - _memoryDevices.Count;
+            IMemoryDevice[] collection = new IMemoryDevice[length];
+            for (int i = 0; i < collection.Length; i++) {
+                collection[i] = memoryDevice;
+            }
+
+            _memoryDevices.AddRange(collection);
         }
-        for (uint i = baseAddress; i < endAddress; i++) {
-            _memoryDevices[i] = memoryDevice;
+        else {
+            for (uint i = baseAddress; i < endAddress; i++) {
+                _memoryDevices[(int)i] = memoryDevice;
+            }
         }
         _devices.Add(new DeviceRegistration(baseAddress, endAddress, memoryDevice));
     }
+    
+    /// <summary>
+    /// Allow a class to unregister for a certain memory range.
+    /// </summary>
+    /// <param name="baseAddress">The start of the frame</param>
+    /// <param name="size">The size of the window</param>
+    /// <param name="memoryDevice">The memory device to use</param>
+    public void UnregisterMapping(uint baseAddress, uint size, IMemoryDevice memoryDevice) {
+        _memoryDevices.RemoveRange((int) baseAddress, (int) size);
```

Comment:
Unregistering addresses can be done by either placing back the regular Ram memory, or by setting them to `null`.
You can't remove them from a list because then all the addresses above it would no longer map to the correct device.

Reply:
Unmapping is supposed to remove the redirection to EMS memory for that EMS page (16 KB within the EMS page frame), so the main memory can be accessed.

Reply:
I'll remove the List and restore the original code in Memory.cs